server/src/main/java/org/opensearch/rest/NamedRoute.java

```diff
@@ -9,17 +9,24 @@
 package org.opensearch.rest;
 
 import org.opensearch.OpenSearchException;
+import org.opensearch.transport.TransportService;
+
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  * A named Route
  *
  * @opensearch.internal
  */
-public class NamedRoute extends RestHandler.Route {
+public class NamedRoute extends RestHandler.Route implements NamedRouteWrapper {
```

Comment:
I think we should switch to a builder pattern, which would look like this, what do you think?  This would allow building on deprecated / replaced easily in the future

```
NamedRoute.createBuilder()
            .Method(GET)
            .Url("/foo/bar")
            .primaryActionName("_plugin/ad/foo-bar-action")
            .additionalActionNames(List.of("legacy-foo-bar"))
            .markAsDeprecated("very stilly indeed")
            .replacedWith(route1);
```

Reply:
done.